src/Mod/Sketcher/Gui/DrawSketchControllableHandler.h

```diff
@@ -0,0 +1,187 @@
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#ifndef SKETCHERGUI_DrawSketchControllableHandler_H
+#define SKETCHERGUI_DrawSketchControllableHandler_H
+
+#include "DrawSketchDefaultHandler.h"
+
+
+namespace SketcherGui
+{
+
+/** @brief Template class intended for handlers controllable by a DrawSketchController
+ *
+ * @details
+ * The template encapsulates a minimal interaction between a DrawSketchDefaultHandler and a
+ * a DrawSketchController, and it is intended to facilitate the creation of enhanced DSHs having
+ * controlling entities such as widgets and on-screen controls.
+ */
+template<typename ControllerT>
+class DrawSketchControllableHandler
+    : public DrawSketchDefaultHandler<typename ControllerT::HandlerType,
+                                      typename ControllerT::SelectModeType,
+                                      ControllerT::AutoConstraintInitialSize,
+                                      typename ControllerT::ContructionMethodType>
+{
+    using HandlerType = typename ControllerT::HandlerType;
+    using SelectModeType = typename ControllerT::SelectModeType;
+    using ConstructionMethodType = typename ControllerT::ContructionMethodType;
+
+    using DSDefaultHandler = DrawSketchDefaultHandler<HandlerType,
+                                                      SelectModeType,
+                                                      ControllerT::AutoConstraintInitialSize,
+                                                      ConstructionMethodType>;
+
+    using ConstructionMachine = ConstructionMethodMachine<ConstructionMethodType>;
+
+    friend ControllerT;
+
+public:
+    DrawSketchControllableHandler(
+        ConstructionMethodType constructionmethod = static_cast<ConstructionMethodType>(0))
+        : DSDefaultHandler(constructionmethod)
+        , toolWidgetManager(static_cast<HandlerType*>(this))
+    {}
+
+    virtual ~DrawSketchControllableHandler() = default;
+
+    /** @name functions NOT intended for specialisation */
+    //@{
+    virtual void mouseMove(Base::Vector2d onSketchPos) override
+    {

```

Comment:
I told you the story of virtual in another comment. The current guidelines are:
1. In the base class, it needs to be virtual and cannot be override/final.
2. In all derived classes, virtual needs to be removed and override or final used